### ISSUE TITLE
last status stack icon can sometimes stick around

### ISSUE THREAD

**Issue by mysterymagination** [mysterymagination/adventures-of-mooty-wort#19]
occasionally I've seen stun's lightning bolt or temper's sword stick around forever.  In the case of sword, I think I noticed an oddment where a sword didn't disappear at the top of a round as expected and then the last icon stuck around forever

**Comment by mysterymagination** [comment]
haven't found a reliable way to repro this, seems random.  However, I was able to see that when it occurred the breakpoints and logs indicated that the last icon should have been removed with no adds back thereafter...  my best guess is something something async, but I have no idea what

**Comment by mysterymagination** [comment]
It seems much more common at the beginning of a fresh game session, when all assets need to load in and nothing is cached.

here's a log set from when it happened right away, with static bolt being the first abl use after game load:
```
alchemy.js:19 status effect unset
mootywortrpgmech.js:219 XHR status says 200
mootywortrpgmech.js:480 stun has 2 ticks remaining
mootywortrpgmech.js:489 status stack; no stack div to remove for id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 2 ticks left
mootywortrpgmech.js:289 GET http://localhost:8080/MoleLauncher/data/fx/pass_stunned.json 404
showSpellEffect @ mootywortrpgmech.js:289
stepCombat @ mootywortrpgmech.js:167
combatLoop @ mootywortrpgmech.js:77
async function (async)
combatLoop @ mootywortrpgmech.js:76
handlePlayerTurnComplete @ mootywortrpgmech.js:949
(anonymous) @ mootywortrpgmech.js:839
painAnimFn @ mootywortrpgmech.js:437
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
mootywortrpgmech.js:219 XHR status says 404
mootywortrpgmech.js:480 stun has 2 ticks remaining
mootywortrpgmech.js:489 status stack; no stack div to remove for id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 2 ticks left
combat.js:109 ticking down The Yawning God's Stun to 1
combat.js:117 The Yawning God takes no action due to his limbs being paralyzed by corruscating electricity!
mootywortrpgmech.js:480 stun has 1 ticks remaining
mootywortrpgmech.js:489 status stack; no stack div to remove for id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 1 ticks left
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 1 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 1 ticks remaining
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 1 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 1 ticks remaining
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 1 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 1 ticks remaining
mootywortrpgmech.js:219 XHR status says 200
mootywortrpgmech.js:480 stun has 1 ticks remaining
mootywortrpgmech.js:485 status stack; removing stack div [object HTMLDivElement] with id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 1 ticks left
mootywortrpgmech.js:289 GET http://localhost:8080/MoleLauncher/data/fx/pass_stunned.json 404
showSpellEffect @ mootywortrpgmech.js:289
stepCombat @ mootywortrpgmech.js:167
combatLoop @ mootywortrpgmech.js:77
async function (async)
combatLoop @ mootywortrpgmech.js:76
handlePlayerTurnComplete @ mootywortrpgmech.js:949
(anonymous) @ mootywortrpgmech.js:839
painAnimFn @ mootywortrpgmech.js:437
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 1 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 1 ticks remaining
mootywortrpgmech.js:219 XHR status says 404
mootywortrpgmech.js:480 stun has 1 ticks remaining
mootywortrpgmech.js:485 status stack; removing stack div [object HTMLDivElement] with id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 1 ticks left
combat.js:109 ticking down The Yawning God's Stun to 0
alchemy.js:22 status effect reversal unset
characters.js:508 The Yawning God's chosen abl is Pestilence
mootywortrpgmech.js:480 stun has 0 ticks remaining
mootywortrpgmech.js:485 status stack; removing stack div [object HTMLDivElement] with id yawning_god_stun_icon_stack
mootywortrpgmech.js:755 status stack; removing status stun from The Yawning God
mootywortrpgmech.js:532 status stack; status effect icon for stun is not needed as the effect has expired
```

compared to when it doesn't happen:
```
alchemy.js:19 status effect unset
mootywortrpgmech.js:219 XHR status says 200
mootywortrpgmech.js:480 stun has 2 ticks remaining
mootywortrpgmech.js:489 status stack; no stack div to remove for id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 2 ticks left
mootywortrpgmech.js:289 GET http://localhost:8080/MoleLauncher/data/fx/pass_stunned.json 404
showSpellEffect @ mootywortrpgmech.js:289
stepCombat @ mootywortrpgmech.js:167
combatLoop @ mootywortrpgmech.js:77
async function (async)
combatLoop @ mootywortrpgmech.js:76
handlePlayerTurnComplete @ mootywortrpgmech.js:949
(anonymous) @ mootywortrpgmech.js:839
painAnimFn @ mootywortrpgmech.js:437
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 2 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 2 ticks remaining
mootywortrpgmech.js:529 status stack; drawing duration index 1 of status effect stack for stun with 2 ticks remaining
mootywortrpgmech.js:219 XHR status says 404
mootywortrpgmech.js:480 stun has 2 ticks remaining
mootywortrpgmech.js:485 status stack; removing stack div [object HTMLDivElement] with id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 2 ticks left
combat.js:109 ticking down The Yawning God's Stun to 1
combat.js:117 The Yawning God takes no action due to his limbs being paralyzed by corruscating electricity!
mootywortrpgmech.js:480 stun has 1 ticks remaining
mootywortrpgmech.js:489 status stack; no stack div to remove for id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 1 ticks left
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 1 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 1 ticks remaining
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 1 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 1 ticks remaining
mootywortrpgmech.js:219 XHR status says 200
mootywortrpgmech.js:480 stun has 1 ticks remaining
mootywortrpgmech.js:485 status stack; removing stack div [object HTMLDivElement] with id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 1 ticks left
mootywortrpgmech.js:289 GET http://localhost:8080/MoleLauncher/data/fx/pass_stunned.json 404
showSpellEffect @ mootywortrpgmech.js:289
stepCombat @ mootywortrpgmech.js:167
combatLoop @ mootywortrpgmech.js:77
async function (async)
combatLoop @ mootywortrpgmech.js:76
handlePlayerTurnComplete @ mootywortrpgmech.js:949
(anonymous) @ mootywortrpgmech.js:839
painAnimFn @ mootywortrpgmech.js:437
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
requestAnimationFrame (async)
painAnimFn @ mootywortrpgmech.js:444
mootywortrpgmech.js:508 status stack; adding stackdiv for stun, who still has 1 ticks left, with offset dimens 24x274 to targetcanvascontainer with offset dimens 241x274
mootywortrpgmech.js:514 status stack; effect with id stunwas found at 0 of the status effect array
mootywortrpgmech.js:516 status stack; stack div left is 0px based on offsetWidth 24 * effect index 0
mootywortrpgmech.js:529 status stack; drawing duration index 0 of status effect stack for stun with 1 ticks remaining
mootywortrpgmech.js:219 XHR status says 404
mootywortrpgmech.js:480 stun has 1 ticks remaining
mootywortrpgmech.js:485 status stack; removing stack div [object HTMLDivElement] with id yawning_god_stun_icon_stack
mootywortrpgmech.js:494 status stack; preparing to add stackdiv for stun, who still has 1 ticks left
combat.js:109 ticking down The Yawning God's Stun to 0
alchemy.js:22 status effect reversal unset
characters.js:508 The Yawning God's chosen abl is Dark Star
mootywortrpgmech.js:480 stun has 0 ticks remaining
mootywortrpgmech.js:485 status stack; removing stack div [object HTMLDivElement] with id yawning_god_stun_icon_stack
mootywortrpgmech.js:755 status stack; removing status stun from The Yawning God
mootywortrpgmech.js:532 status stack; status effect icon for stun is not needed as the effect has expired
```

**Comment by mysterymagination** [comment]
hmm, from the trace above it looks like the problem may be multiple adds basically layering dup stack divs with the same id on top of each other and we somehow wind up with at least one more add than remove.  Can you add multiple elements to the DOM with the same id?  I thought the whole point of id was to be unique...
Edit: ye can totally fookin do that and with the right position props they will happily lay out on top of each other.  Goddammit, DOM.

Anyway, that's likely the cause and at some point you can probably even see the dup div in the dom ;)

**Comment by mysterymagination** [comment]
confirmed by checking for a non-null element of the id we want to use right before creating a new one -- and by seeing multiple stack divs with the same stun id in the dom inspector X|

**Comment by mysterymagination** [comment]
fixed in ffabc453e123fe672a567c039b0ffce1210710de via ensuring we have a chance to remove any existing stack div of the given id before adding a new one in the image cb
